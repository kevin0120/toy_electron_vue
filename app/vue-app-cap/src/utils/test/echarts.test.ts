import { describe, expect, it } from 'vitest';
import {
  type tFormatterParams,
  tooltipFormatter,
  seriesLineFormatter,
  isShowTorqueAngle,
  isShowAngle,
  isShowTorque,
  getMarkLine,
  convertTo2DArray,
  getMarkPointData
} from '../echarts';
import type { tProbaResult } from '/@/api/types/curves';
import type { tSeriesData } from '/@/types/chart';

describe('Test tooltipFormatter function', () => {
  const params: tFormatterParams<any>[] = [
    {
      componentType: 'series',
      seriesType: 'bar',
      seriesIndex: 0,
      seriesName: '参考曲线',
      name: 'Milk Tea',
      dataIndex: 1,
      value: [12, 800],
      encode: {},
      dimensionNames: [],
      dimensionIndex: 0,
      data: { error_code: 'Milk Tea', count: 83.1 },
      color: '#5470c6'
    },
    {
      componentType: 'series',
      seriesType: 'bar',
      seriesIndex: 0,
      seriesName: '参考曲线',
      name: 'Milk Tea',
      dataIndex: 1,
      value: [12, 802],
      encode: {},
      dimensionNames: [],
      dimensionIndex: 0,
      data: { error_code: 'Milk Tea', count: 83.1 },
      color: '#5470c6'
    }
  ];
  const customToolTipInfoList = [
    [
      {
        key: '最大扭矩',
        value: 12
      }
    ]
  ];
  it('Return value snapshot when a parameter is passed in', () => {
    expect(tooltipFormatter(params)).toMatchSnapshot();
  });
  it('Return value snapshot when two parameters are passed in', () => {
    expect(tooltipFormatter(params, customToolTipInfoList)).toMatchSnapshot();
  });
  it('Return value snapshot when three parameters are passed in', () => {
    expect(tooltipFormatter(params, customToolTipInfoList, true)).toMatchSnapshot();
  });
});

describe('Test seriesLineFormatter function', () => {
  const index = 1;
  const data: tSeriesData = [{ value: [0, 1] }, { value: [1, 1] }, { value: [2, 3] }];
  const name = '测试数据';
  const showArea = true;
  const markLine = {
    x: {
      max: 10,
      target: 8,
      min: 6
    },
    y: {
      max: 10,
      target: 8,
      min: 6
    }
  };
  const markPointData = [{ clamp_angle: 3, clamp_torque: 2, xAxis: 2, yAxis: 3 }];
  it('Ok Snapshot of tightening results basic configuration data', () => {
    expect(seriesLineFormatter(index, data, name, 'ok')).toMatchSnapshot();
  });
  it('Ok Snapshot of tightening results basic configuration data 1', () => {
    expect(seriesLineFormatter(index, data, name, 'ok', showArea)).toMatchSnapshot();
  });
  it('Ok Snapshot of tightening results basic configuration data 2', () => {
    expect(seriesLineFormatter(index, data, name, 'ok', showArea, markLine)).toMatchSnapshot();
  });
  it('Ok Snapshot of tightening results basic configuration data 3', () => {
    expect(seriesLineFormatter(index, data, name, 'ok', showArea, markLine, markPointData)).toMatchSnapshot();
  });
});

describe('Test isShowTorqueAngle , isShowAngle, isShowTorque function', () => {
  it('should ShowTorqueAngle', function () {
    expect(isShowTorqueAngle(0)).toBeTruthy();
  });
  it('should ShowTorque', function () {
    expect(isShowTorque(1)).toBeTruthy();
  });
  it('should ShowAngle', function () {
    expect(isShowAngle(2)).toBeTruthy();
  });
});

describe('Test getMarkLine function', () => {
  const x = {
    max: 10,
    target: 8,
    min: 6
  };
  const y = {
    max: 10,
    target: 8,
    min: 6
  };
  const errIndex = 4;
  it('No data display ', () => {
    expect(getMarkLine(x, y, errIndex)).toBeUndefined();
  });
  it('should toEqual ShowTorqueAngle data', () => {
    expect(getMarkLine(x, y, 0)).toEqual({ x, y });
  });
  it('should toEqual ShowTorque data', () => {
    expect(getMarkLine(x, y, 1)).toEqual({ y });
  });
  it('should toEqual ShowAngle data', () => {
    expect(getMarkLine(x, y, 2)).toEqual({ y });
  });
});

describe('Test convertTo2DArray function', () => {
  const arr1 = [1, 2, 3];
  const arr2 = [4, 5, 6];
  const len = arr1.length;
  it('should have [1,4] ', function () {
    expect(convertTo2DArray(arr1, arr2)).toContainEqual([1, 4]);
  });
  it('toHaveLength', function () {
    expect(convertTo2DArray(arr1, arr2)).toHaveLength(len);
  });
  it('toHaveLength 0', function () {
    expect(convertTo2DArray([], arr2)).toHaveLength(0);
  });
  it('toHaveLength 0', function () {
    expect(convertTo2DArray(arr1, [])).toHaveLength(0);
  });
});

describe('Test getMarkPointData function', () => {
  const arr1: Array<[number, number]> = [
    [1, 1],
    [2, 2],
    [3, 3]
  ];
  const data: tProbaResult = {
    analysis_result_state: 'ok',
    entity_id: '123456',
    clamp_torque: 1,
    clamp_angle: 1,
    snug_index: 2,
    snug_angle: 1,
    snug_torque: 1
  };
  it('should toContainEqual', function () {
    expect(getMarkPointData(data, arr1)).toContainEqual({
      clamp_angle: 1,
      clamp_torque: 1,
      xAxis: 3,
      yAxis: 3
    });
  });
});
