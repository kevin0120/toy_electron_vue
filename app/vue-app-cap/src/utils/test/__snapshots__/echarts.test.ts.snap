// Vitest Snapshot v1

exports[`Test seriesLineFormatter function > Ok Snapshot of tightening results basic configuration data 1`] = `
{
  "color": "#4DD07D",
  "data": [
    [
      0,
      1,
    ],
    [
      1,
      1,
    ],
    [
      2,
      3,
    ],
  ],
  "index": 1,
  "name": "测试数据",
  "showSymbol": false,
  "smooth": true,
  "type": "line",
}
`;

exports[`Test seriesLineFormatter function > Ok Snapshot of tightening results basic configuration data 1 1`] = `
{
  "areaStyle": {
    "color": LinearGradient {
      "colorStops": [
        {
          "color": "#4DD07D",
          "offset": 0,
        },
        {
          "color": "#fff",
          "offset": 1,
        },
      ],
      "global": false,
      "type": "linear",
      "x": 0,
      "x2": 0,
      "y": 0,
      "y2": 1,
    },
  },
  "color": "#4DD07D",
  "data": [
    [
      0,
      1,
    ],
    [
      1,
      1,
    ],
    [
      2,
      3,
    ],
  ],
  "index": 1,
  "name": "测试数据",
  "showSymbol": false,
  "smooth": true,
  "type": "line",
}
`;

exports[`Test seriesLineFormatter function > Ok Snapshot of tightening results basic configuration data 2 1`] = `
{
  "areaStyle": {
    "color": LinearGradient {
      "colorStops": [
        {
          "color": "#4DD07D",
          "offset": 0,
        },
        {
          "color": "#fff",
          "offset": 1,
        },
      ],
      "global": false,
      "type": "linear",
      "x": 0,
      "x2": 0,
      "y": 0,
      "y2": 1,
    },
  },
  "color": "#4DD07D",
  "data": [
    [
      0,
      1,
    ],
    [
      1,
      1,
    ],
    [
      2,
      3,
    ],
  ],
  "index": 1,
  "markLine": {
    "animation": false,
    "data": [
      {
        "label": {
          "fontSize": 12,
          "formatter": "最大 {c}",
          "position": "insideEndTop",
        },
        "lineStyle": {
          "color": "#CC8925",
          "type": "solid",
          "width": 1,
        },
        "name": "max",
        "xAxis": 10,
      },
      {
        "label": {
          "fontSize": 12,
          "formatter": "目标 {c}",
          "position": "insideEndTop",
        },
        "lineStyle": {
          "color": "#CC8925",
          "type": "dashed",
          "width": 2,
        },
        "name": "target",
        "xAxis": 8,
      },
      {
        "label": {
          "fontSize": 12,
          "formatter": "最小 {c}",
          "position": "insideEndTop",
        },
        "lineStyle": {
          "color": "#CC8925",
          "type": "solid",
          "width": 1,
        },
        "name": "min",
        "xAxis": 6,
      },
      {
        "label": {
          "fontSize": 12,
          "formatter": "最大 {c}",
          "position": "insideEndTop",
        },
        "lineStyle": {
          "color": "#CC8925",
          "type": "solid",
          "width": 1,
        },
        "name": "max",
        "yAxis": 10,
      },
      {
        "label": {
          "fontSize": 12,
          "formatter": "目标 {c}",
          "position": "insideEndTop",
        },
        "lineStyle": {
          "color": "#CC8925",
          "type": "dashed",
          "width": 2,
        },
        "name": "target",
        "yAxis": 8,
      },
      {
        "label": {
          "fontSize": 12,
          "formatter": "最小 {c}",
          "position": "insideEndTop",
        },
        "lineStyle": {
          "color": "#CC8925",
          "type": "solid",
          "width": 1,
        },
        "name": "min",
        "yAxis": 6,
      },
    ],
    "label": {
      "show": true,
    },
    "silent": true,
    "symbol": "none",
  },
  "name": "测试数据",
  "showSymbol": false,
  "smooth": true,
  "type": "line",
}
`;

exports[`Test seriesLineFormatter function > Ok Snapshot of tightening results basic configuration data 3 1`] = `
{
  "areaStyle": {
    "color": LinearGradient {
      "colorStops": [
        {
          "color": "#4DD07D",
          "offset": 0,
        },
        {
          "color": "#fff",
          "offset": 1,
        },
      ],
      "global": false,
      "type": "linear",
      "x": 0,
      "x2": 0,
      "y": 0,
      "y2": 1,
    },
  },
  "color": "#4DD07D",
  "data": [
    [
      0,
      1,
    ],
    [
      1,
      1,
    ],
    [
      2,
      3,
    ],
  ],
  "index": 1,
  "markLine": {
    "animation": false,
    "data": [
      {
        "label": {
          "fontSize": 12,
          "formatter": "最大 {c}",
          "position": "insideEndTop",
        },
        "lineStyle": {
          "color": "#CC8925",
          "type": "solid",
          "width": 1,
        },
        "name": "max",
        "xAxis": 10,
      },
      {
        "label": {
          "fontSize": 12,
          "formatter": "目标 {c}",
          "position": "insideEndTop",
        },
        "lineStyle": {
          "color": "#CC8925",
          "type": "dashed",
          "width": 2,
        },
        "name": "target",
        "xAxis": 8,
      },
      {
        "label": {
          "fontSize": 12,
          "formatter": "最小 {c}",
          "position": "insideEndTop",
        },
        "lineStyle": {
          "color": "#CC8925",
          "type": "solid",
          "width": 1,
        },
        "name": "min",
        "xAxis": 6,
      },
      {
        "label": {
          "fontSize": 12,
          "formatter": "最大 {c}",
          "position": "insideEndTop",
        },
        "lineStyle": {
          "color": "#CC8925",
          "type": "solid",
          "width": 1,
        },
        "name": "max",
        "yAxis": 10,
      },
      {
        "label": {
          "fontSize": 12,
          "formatter": "目标 {c}",
          "position": "insideEndTop",
        },
        "lineStyle": {
          "color": "#CC8925",
          "type": "dashed",
          "width": 2,
        },
        "name": "target",
        "yAxis": 8,
      },
      {
        "label": {
          "fontSize": 12,
          "formatter": "最小 {c}",
          "position": "insideEndTop",
        },
        "lineStyle": {
          "color": "#CC8925",
          "type": "solid",
          "width": 1,
        },
        "name": "min",
        "yAxis": 6,
      },
    ],
    "label": {
      "show": true,
    },
    "silent": true,
    "symbol": "none",
  },
  "markPoint": {
    "data": [
      {
        "clamp_angle": 3,
        "clamp_torque": 2,
        "xAxis": 2,
        "yAxis": 3,
      },
    ],
    "itemStyle": {
      "color": "#fff",
      "shadowBlur": 2,
      "shadowColor": "rgba(0, 0, 0, 0.5)",
    },
    "label": {
      "fontSize": 14,
      "formatter": [Function],
      "offset": [
        0,
        5,
      ],
      "position": "insideTop",
      "rich": {
        "label": {
          "fontSize": 14,
          "fontWeight": "bold",
          "height": 20,
        },
      },
      "show": true,
    },
    "symbol": "path://M14,0 L132,0 C139.731986,-1.42034288e-15 146,6.2680135 146,14 L146,63 C146,70.7319865 139.731986,77 132,77 L77,77 L77,77 L73,85 L69,77 L14,77 C6.2680135,77 9.46895252e-16,70.7319865 0,63 L0,14 C-9.46895252e-16,6.2680135 6.2680135,1.42034288e-15 14,0 Z",
    "symbolKeepAspect": true,
    "symbolOffset": [
      0,
      "-30%",
    ],
    "symbolSize": 190,
  },
  "name": "测试数据",
  "showSymbol": false,
  "smooth": true,
  "type": "line",
}
`;

exports[`Test tooltipFormatter function > Return value snapshot when a parameter is passed in 1`] = `
"
           <div style=\\"display: flex; align-items: center; width: 100%\\">
             <div style=\\"background: #5470c6; height: 10px; width: 10px; border-radius: 50%\\"></div>
             <div style=\\"font-weight: 700\\">参考曲线</div>
           </div>
            扭矩: 800 NM
      <br />角度: 12 Deg<br />
           <div style=\\"display: flex; align-items: center; width: 100%\\">
             <div style=\\"background: #5470c6; height: 10px; width: 10px; border-radius: 50%\\"></div>
             <div style=\\"font-weight: 700\\">参考曲线</div>
           </div>
            扭矩: 802 NM
      <br />角度: 12 Deg<br />"
`;

exports[`Test tooltipFormatter function > Return value snapshot when three parameters are passed in 1`] = `
"
           <div style=\\"display: flex; align-items: center; width: 100%\\">
             <div style=\\"background: #5470c6; height: 10px; width: 10px; border-radius: 50%\\"></div>
             <div style=\\"font-weight: 700\\">参考曲线</div>
           </div>
            扭矩: 800 NM
      <br />角度: 12 Deg<br />最大扭矩: 12<br />
           <div style=\\"display: flex; align-items: center; width: 100%\\">
             <div style=\\"background: #5470c6; height: 10px; width: 10px; border-radius: 50%\\"></div>
             <div style=\\"font-weight: 700\\">参考曲线</div>
           </div>
            扭矩: 802 NM
      <br />角度: 12 Deg<br /><br /><div style=\\"color: #f6d544\\">均值: 801</div>
                  <div style=\\"color: #f6d544\\">标准差: 1</div><br />"
`;

exports[`Test tooltipFormatter function > Return value snapshot when two parameters are passed in 1`] = `
"
           <div style=\\"display: flex; align-items: center; width: 100%\\">
             <div style=\\"background: #5470c6; height: 10px; width: 10px; border-radius: 50%\\"></div>
             <div style=\\"font-weight: 700\\">参考曲线</div>
           </div>
            扭矩: 800 NM
      <br />角度: 12 Deg<br />最大扭矩: 12<br />
           <div style=\\"display: flex; align-items: center; width: 100%\\">
             <div style=\\"background: #5470c6; height: 10px; width: 10px; border-radius: 50%\\"></div>
             <div style=\\"font-weight: 700\\">参考曲线</div>
           </div>
            扭矩: 802 NM
      <br />角度: 12 Deg<br />"
`;
